### ISSUE TITLE
Improve devfile editing UX

### ISSUE THREAD

**Issue by l0rd** [eclipse/che#21606]
### Is your enhancement related to a problem? Please describe

To simplify the day 1 activities (start a sample workspace, start a workspace from a git repository) we have removed the Devfile editor from the Dashboard as editing a YAML file to create a workspace was a daunting activity for new users.

As a consequence we have sacrificed a day 2 activity, something that more experienced users loved: the devfile editor. From devfile editor in the dashboard allowed to create and edit a devfile (re)start a workspace from it.

### Describe the solution you'd like

There will be multiple alternatives:
- [Start from no-devfile](#21340) and [progressively build one](#18670) [without editing one line of YAML](#18671)
- [Follow a link to inspect and edit the DevWorkspace object in OpenShift Console](https://github.com/eclipse/che/issues/22422)
- Advanced Devfile editing experience in VS Code with code completion with the [YAML VS Code extension](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml)

The progressive editing flow is the following one:
1. start a workspace without Devfile: empty or from a git repository (passed as a URL parameter)
2. in the IDE it will be possible to use a GUI tool to generate or to edit the devfile (no YAML)
3. from the IDE it will be possible to restart a workspace from the created / modified devfile
4. for advanced configuration it will be possible to edit a Devfile YAML with code completion and API documentation in the IDE itself

Subtasks:
- https://github.com/eclipse/che/issues/21554
- https://github.com/eclipse/che/issues/20738
- https://github.com/eclipse/che/issues/21340
- https://github.com/eclipse/che/issues/21184
- https://github.com/eclipse/che/issues/18670 
- https://github.com/eclipse/che/issues/22422
- https://github.com/eclipse/che/issues/22115
- https://github.com/eclipse/che/issues/22228
- https://github.com/eclipse/che/issues/18671
- https://github.com/eclipse/che/issues/22612
- https://github.com/eclipse/che/issues/22640

### Describe alternatives you've considered

We want to avoid the re-introduction of the user dashboard devfile editor because we want that to happen [in the IDE](https://github.com/eclipse/che/issues/18670). That's where a developer has the best development experience and that's where they expect to edit their workspace. Having to go back to the dashboard to edit the workspace is awkward. This is a feedback we got recurrently from users.

### Additional context

https://issues.redhat.com/browse/CRW-3152

**Comment by RickJWagner** [comment]
Thank you for considering these important usability issues.
There is a 'catch-22' to consider.  If we want to use the IDE to edit the devfile, we need to consider the case of the user that has an invalid devfile (which will prevent starting of the workspace.)

Example:  A user wants to develop a new piece of software.  They may start with some minimal set of source code, maybe copied from a blog, previous project or generated by something like spring initializer.  Any way they do it, they start with some source code, which they probably commit to git.
At this point, we are in the 'problem zone'.  The user cannot start a workspace-- they have no devfile.  They could take an existing one and add it to the git repo, but the devfile must be perfect (without benefit of some UI to validate it!) or the dev workspace will not start.  The cycle to resolve the issue is:
1) Edit the devfile without an IDE
2) Commit the change
3) Try again to start the dev space,  hoping this time it will succeed and provide the IDE that will allow editing the devfile.  (Repeat as needed).

One possibility:  Provide a known minimal devfile that is very robust and should almost always be able to start a dev space.  In the above scenario, the user would add this 'bulletproof devfile' to the project along with the initial source.  The user could then start the dev space and use the available IDE to edit the 'bulletproof' devfile, transforming it into whatever is needed.

There are probably other good paths forward also.  Just please keep in mind that we should be able to edit the devfile, even when the devfile has not yet been validated by Che / Dev Spaces.

**Comment by l0rd** [comment]
@RickJWagner what about https://github.com/eclipse/che/issues/20738 and 

![warning-invalid-devfile](https://user-images.githubusercontent.com/606959/182797889-457e8f6d-2654-4d35-a776-8798b63f823b.png)

**Comment by RickJWagner** [comment]
Yes, thanks for that @l0rd .  I hadn't seen #20738, that will help nicely.

**Comment by nickboldt** [comment]
any further progress on this? we were at 5 of 7 tasks complete in December... should the other two be raised as priority for the next sprint?

**Comment by l0rd** [comment]
Both remaining issues are [in Team B sprint backlog](https://github.com/eclipse/che/issues/21942)

**Comment by ibuziuk** [comment]
Exluding JetBrains from the epic - https://github.com/eclipse/che/issues/22487  and adding to https://issues.redhat.com/browse/CRW-3398
We can resolve the issue once we publish the walkthrough extension and add it by default to Dev Spaces

**Comment by ibuziuk** [comment]
@vitaliy-guliy I believe we can close the epic once vscode walkthrough extension is added by default to che-code

**Comment by vitaliy-guliy** [comment]
A pull request that includes the extension into built-ins https://github.com/che-incubator/che-code/pull/326
I would like to have the `latest` tag for the extension at open-vsix. It will allow us to update and publish a new version of the extension without updating the reference in che-code.

**Comment by vitaliy-guliy** [comment]
> `latest` should work. Does it fail for you?

It's strange I missed it.

But it seems vs code builder wants to have a precise version
```
2283.6 [14:14:39] Error: Request https://open-vsx.org/vscode/gallery/publishers/devfile/vsextensions/vscode-devfile/latest/vspackage failed with status code: 404
```
With `0.0.2` the following request is succeeded
```
https://open-vsx.org/vscode/gallery/publishers/devfile/vsextensions/vscode-devfile/0.0.2/vspackage
```

**Comment by vitaliy-guliy** [comment]
There is a regression in VS Code that made not possible to add a container endpoint, environment variable and a command
https://github.com/microsoft/vscode/issues/204522

From the good news, the issue was added to February milestone and should be fixed soon.

**Comment by ibuziuk** [comment]
@vitaliy-guliy the issue you opened has been autoclosed. is it fixed now in 7.82.0 - https://github.com/microsoft/vscode/issues/204522

**Comment by vitaliy-guliy** [comment]
> @vitaliy-guliy the issue you opened has been autoclosed. is it fixed now in 7.82.0 - [microsoft/vscode#204522](https://github.com/microsoft/vscode/issues/204522)

Have just checked it on Dev Spaces 3.12 with Che Code 7.82.1 (VS Code 1.87.0) and can confirm the bug is still there.

Despite the [issue](https://github.com/microsoft/vscode/issues/204522) is closed, as not reproducible on Gitpod with VS Code 1.87.0, the bug is actual for GitHub Codespaces with VS Code 1.85.2

**Comment by vitaliy-guliy** [comment]
The extension is working fine in che-code 7.84.0, bug with popups was fixed in the upstream.
